crates/aptos-rosetta/src/account.rs

```diff
@@ -172,7 +179,171 @@ async fn get_balances(
+                                        contract.pool_address, result
+                                    )
+                                },
+                            }
+                        }
+
+                        if let Some(balance) = total_active {
+                            balances.push(Amount {
+                                value: balance.to_string(),
+                                currency: native_coin(),
+                            })
+                        }
+                    } else if account.is_pending_inactive_stake() {
+                        // For total pending inactive stake, collect all underlying staking contracts and combine
+                        let mut total_pending_inactive: Option<u64> = None;
+                        maybe_operators = Some(vec![]);
+                        for (operator, contract) in store.staking_contracts {
+                            // Keep track of operators
+                            maybe_operators.as_mut().unwrap().push(operator);
+                            match get_stake_balances(
+                                rest_client,
+                                &account,
+                                contract.pool_address,
+                                version,
+                            )
+                            .await
+                            {
+                                Ok(Some(stake_balances)) => {
+                                    if let Ok(balance) = stake_balances.pending_inactive {
+                                        total_pending_inactive = Some(
+                                            total_pending_inactive.unwrap_or_default()
+                                                + u64::from_str(&balance.value).unwrap_or_default(),
+                                        );
+                                    } else {
+                                        warn!("Failed to retrieve pending inactive stake for {}", contract.pool_address) 
+                                    }
+                                },
+                                result => {
+                                    warn!(
+                                        "Failed to retrieve balances for {}: {:?}",
+                                        contract.pool_address, result
+                                    )
+                                },
+                            }
+                        }
+
+                        if let Some(balance) = total_pending_inactive {
+                            balances.push(Amount {
+                                value: balance.to_string(),
+                                currency: native_coin(),
+                            })
+                        }
+                    } else if account.is_inactive_stake() {
+                        // For total inactive stake, collect all underlying staking contracts and combine
```

Comment:
Small nits:
- Looks like multiple `else if` can be replace with the `match`. But not sure if that makes the code looks worse if multiple `match` exist in different scope and feel free to push back.
- These four types balances looks very similar in terms of looping contracts and `get_stake_balances` so we can merge them together to reduce duplicate code and only differentiate the desire field in  `get_stake_balances` result?

Reply:
yea i tried multiple match in different scope, but it looked very hard to read.
good idea, i'll differentiate desired field in `get_stake_balances`